riscv-overlay-software-standard-draft.adoc

```diff
@@ -263,44 +304,97 @@ The descriptor will be materialized at link time.
-.  Changes in the debugger should be generic in such a way that all related _“RT-Eng-Debugger-handshake”_  will be in an external file to hook into the debugger.
-. We shall have debug information for overlay functions, that information should be symmetric if a function is placed in several groups (*_multi group_*).
+. Overlay RT-Engine awareness:
+.. For backtracing, the debugger should be able to unwind the stack with awareness of calls/returns through the RT-Engine.
+.. To give a comfortable debugging experience we should have an option to “skip” through the RT-Engine when doing a step on a function call or return. E.g. if we step at call to function myOverlayFoo(), the debugger should skip through the RT-Engine and the next PC we see will be the beginning of myOverlayFoo() and not within the RT-Engine. Similarly, if we step at a function return, the debugger should skip through the RT-Engine and the next PC we see will be at the return address in the caller.
+.. There should also be an option to disable this “skip” functionality to allow debugging of the RT-Engine.
+. The RT-Engine will have three anchors in the source code for debugger-engine synchronization.
+With those anchors, the debugger will be able to sync with RT-Engine logic-flow.
+Those anchors manifest by symbols and break-points: entering, exiting, data-base-sync-point.
++
+[[RT-Engine-and-Debugger-sync-anchors]]
+.RT-Engine and Debugger sync anchors
+[%header,cols="7a,20a"]
+|===
+| *Sync point* | *Description*
+|Enter RT-Engine| The entry point to the RT-Engine
+|Exit RT-Engine| The exit point from the RT-Engine
+|Data-base-sync-point| The sync point on which the RT-Engine refresh the loaded (mapping update) groups
+|===
+
+. The debugger will be agnostic to the existence of a RTOS, this means a context switch can happen during an overlay operation and the debugger should hold a valid sequence.
+. Changes in the debugger should be generic in such a way that all related _“RT-Eng-Debugger-handshake”_  will be in an external file to hook into the debugger.
+. We shall have debug information for overlay functions and overlay data. That information should be symmetric if a function is placed in several groups (*_multi group_*).
+
+{nbsp} +
+
+
+[[Appendix-A-Grouping-file-syntax]]
+= Appendix A - Grouping file syntax
+
+The linker can receive an input file to give it details about assignments of groups to functions.
+e.g. myFunction() should be in group 1.
+This file should be in comma-separate-value syntax (CSV), as described:
+
+*	Each new line represent a function
+*	First column holds a function name
+*	Each next column holds a group number to assign the function
+
+[[Example-for-Grouping-file-CSV]]
+.Example for Grouping file CSV
+
+[%header, format=csv]
+|===
+*Function name* , *Group number*, *Group number*, *Group ...*
```

Comment:
Will the header be required or optional?

Reply:
Good question. let's have a header, and on the implementation side, we will check if there is a header or not (to support both cases)